parlai/tasks/funpedia/agents.py

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from parlai.core.teachers import FbDialogTeacher, FixedDialogTeacher
+from .build import build
+
+import copy
+import os
+
+"""
+Task for rephrasing sentences from Wikipedia conditioned on a persona.
+"""
+
+
+def _path(opt):
+    build(opt)
+    datatype = opt['datatype'].split(':')[0]
+    return os.path.join(
+        opt['datapath'],
+        'rephrase_sentences',
+        'rephrase_sentences_' + datatype + '_0703.txt',
+    )
+
+
+def _choose_sentence_path(opt):
+    build(opt)
+    datatype = opt['datatype'].split(':')[0]
+    return os.path.join(
+        opt['datapath'], 'rephrase_sentences', 'choose_sentence_' + datatype + '.txt'
+    )
+
+
+def _strip_reader(filename):
+    """
+    Reads a file, stripping line endings.
+    """
+    with open(filename) as f:
+        for line in f:
+            yield line.rstrip()
+
+
+# this is a standard itertools recipe, but not included by default.
+# see https://docs.python.org/3/library/itertools.html#itertools-recipes
```

Comment:
from this page looks like we can just do `pip install more-itertools` to import this `grouper` function

Reply:
do you think we should add this as a requirement or leave as-is for now?

Reply:
can just leave as-is, not super important

Reply:
maybe make a note that it's there if people want to use it